packages/eslint-config-vtex/rules/tests.js

```diff
@@ -0,0 +1,75 @@
+// Jest: https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules
+// Cypress: https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules
+module.exports = {
+  overrides: [
+    // ! CYPRESS
+    {
+      files: ['**/cypress/**/*{.test,.spec,_spec}.{ts,tsx,js,jsx}'],
+      extends: ['plugin:cypress/recommended'],
+      rules: {
+        // Enforce assertions before taking a screenshot
+        // https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/assertion-before-screenshot.md
+        'cypress/assertion-before-screenshot': 'warn',
+      },
+    },
+    // ! JEST
+    {
+      // Run through every test file found
+      files: ['*.{test,spec}.{ts,tsx,js,jsx}'],
+      // Unless it's inside a cypress directory
+      excludedFiles: ['**/cypress/**'],
+      extends: ['plugin:jest/recommended', 'plugin:jest/style'],
+      rules: {
+        // Enforce consistent a test method name
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/consistent-test-it.md
+        'jest/consistent-test-it': [
+          'warn',
+          {
+            fn: 'test',
+            withinDescribe: 'it',
+          },
+        ],
+
+        // Disallow alias methods
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-alias-methods.md
+        // TODO: enable?
+        'jest/no-alias-methods': 'off',
+
+        // Disallow duplicate setup/teardown hooks
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-duplicate-hooks.md
+        'jest/no-duplicate-hooks': 'error',
+
+        // Disallow using expect().resolves
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-expect-resolves.md
+        'jest/no-expect-resolves': 'error',
+
+        // Suggest to have all hooks at top-level before tests
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/prefer-hooks-on-top.md
+        'jest/prefer-hooks-on-top': 'error',
+
+        // Suggest jest.spyOn() instead of jest.fn()
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/prefer-spy-on.md
+        'jest/prefer-spy-on': 'warn',
+
+        // Suggest using test.todo()
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/prefer-todo.md
+        'jest/prefer-todo': 'warn',
+
+        // Disallow conditional logic inside tests
+        // https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-if.md
+        // TODO: enable?
+        'jest/no-if': 'off',
```

Comment:
I think conditionals inside tests are useful in a case like

```js
describe.each`
  accountName          | env
  ${'storecomponents'} | ${'stable'}
  ${'checkoutio'}      | ${'stable'}
`('Foo component test', ({ accountName, env }) => {
  it(`should do something on ${accountName}-${env}`, () => {
    if (accountName === 'storecomponents') {
      // do something specific to this account
    }
  })
})
```

Reply:
sad feelings only

Reply:
:mind-blown:

Reply:
Didn't know about `describe.each`, but this could be done as a iteration of a simple array instead 🤓 

```js
const foo = [['storecomponents', 'stable'],['checkoutio','stable']]
foo.forEach(([accountName, env]) => {
  describe('Foo component test', () => { ... })
})
```